### PR TITLE
Replace the call to hawkey with a call to DNF in repo sanity check

### PR DESCRIPTION
**Pull request by puiterwijk**
libdnf/hawkey silently dropped the API we were using to do repo sanity checks.
So instead, let's just shell out to DNF and make it their problem.

Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

File: bodhi/server/util.py [hunk 1]
```diff
@@ -234,6 +235,7 @@ def get_critpath_components(collection='master', component_type='rpm', component
     return critpath_components
 
 
+
```
Comment (bowlofeggs) [inline bodhi/server/util.py]: flake8 might get mad about there being 3 lines here instead of 2.

File: bodhi/server/util.py [hunk 2]
```diff
@@ -249,62 +251,110 @@ def sanity_check_repodata(myurl, repo_type):
...
+        # "tests" is a list of tuples with (dnf args, expected output) to run.
+        # For every test, DNF is run with the arguments, and if the expected output is not found,
+        #  an error is raised.
+        tests = []
+
+        if repo_type in ('yum', 'source'):
+            tests.append((['list', 'available'], 'testrepo'))
+        elif repo_type == 'module':
+            tests.append((['module', 'list'], '.*'))
+        # TODO: Add tests that checks for a random build or module and gets its info.
```
Comment (bowlofeggs) [inline bodhi/server/util.py]: Let's drop this and file an issue for it - it's a good idea ☺

File: bodhi/server/util.py [hunk 3]
```diff
@@ -249,62 +251,110 @@ def sanity_check_repodata(myurl, repo_type):
...
+
+        for test in tests:
+            dnfargs, expout = test
+
+            # Make sure every DNF test runs in a new temp dir
+            testdir = tempfile.mkdtemp(dir=tmpdir)
+            output = sanity_check_repodata_dnf(testdir, myurl, *dnfargs)
+            if (expout == ".*" and len(output.strip()) != 0) or (expout in output):
+                return
```
Comment (bowlofeggs) [inline bodhi/server/util.py]: Do you want to return here, or continue?

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

File: bodhi/server/util.py (hunk 1 above)
Comment (puiterwijk) [reply bodhi/server/util.py]: Yep, it did, and I fixed that :).

File: bodhi/server/util.py (hunk 3 above)
Comment (puiterwijk) [reply bodhi/server/util.py]: continue. This was me being silly.

File: bodhi/server/util.py (hunk 2 above)
Comment (puiterwijk) [reply bodhi/server/util.py]: I wanted to do this before I submitted the PR, but didn't get to it yet. I'll file the issue instead and try to do this tomorrow :).

**Review by bowlofeggs — COMMENTED** [review]
Sweet - looks good! I will squash and then let mergify do its thing. Thanks so much!

File: bodhi/server/util.py (hunk 2 above)
Comment (puiterwijk) [reply bodhi/server/util.py]: And issue filed as #3199.

**Review by bowlofeggs — APPROVED** [review]
Thanks so much!

**Comment by puiterwijk** [comment]
The latest change adds the one missing line of diff cover, and fixes some flake8 errors.
I'm hopeful this'll pass CI.

**Comment by puiterwijk** [comment]
Had forgotten to run flake8 over the test :).
Additionally, newline in docblock added.

File: bodhi/server/util.py [hunk 4]
```diff
@@ -249,62 +248,110 @@ def sanity_check_repodata(myurl, repo_type):
...
+    Because DNF does not provide any usable API they won't break in a year time, we just shell out
+    to run some repository actions on it.
+
+    Args:
+        tempdir (basestring): Temporary directory that will be removed at the end.
+        myurl (basestring): A path to a repodata directory.
+        dnf_args (list): A list of arguments after DNF is set up to use the repo under test.
+            Example would be ["search", "kernel"] or ["module", "list"].
+    Raises:
+        Exception: If the repodata is not valid or does not exist.
```
Comment (sebwoj) [inline bodhi/server/util.py:343]: Actually, [function](https://docs.python.org/3/library/subprocess.html#subprocess.check_output) is returning [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError)

**Review by sebwoj — CHANGES_REQUESTED** [review]

File: bodhi/server/util.py (hunk 4 above)
Comment (puiterwijk) [reply bodhi/server/util.py:343]: For now that's the only option, yes. But I have plans to add more checks, which might return other exceptions.
And the current docblock is technically correct:

```
>>> issubclass(subprocess.CalledProcessError, Exception)
True
```

**Comment by bowlofeggs** [comment]
This patch is planned for inclusion in the upcoming 4.0.0 release: #3221

**Comment by bowlofeggs** [comment]
Bodhi 4.0.0 beta is built and deployed to staging:

https://copr.fedorainfracloud.org/coprs/bowlofeggs/bodhi-pre-release
https://bodhi.stg.fedoraproject.org/

**Pull request merged**